File: src/client/java/fr/hardel/asset_editor/client/WorkspaceSyncGateway.java
package fr.hardel.asset_editor.client;

import fr.hardel.asset_editor.network.workspace.WorkspaceElementSnapshot;
import fr.hardel.asset_editor.network.workspace.WorkspaceSyncPayload;
import net.minecraft.resources.Identifier;

import java.util.List;
import java.util.Set;

public interface WorkspaceSyncGateway {
    void handleWorkspaceSync(WorkspaceSyncPayload payload);

    void handlePackWorkspaceSync(String packId, Identifier registryId, List<WorkspaceElementSnapshot> snapshots, Set<Identifier> modifiedIds);
}

File: src/client/kotlin/fr/hardel/asset_editor/client/compose/lib/action/WorkspaceSyncGateway.kt
package fr.hardel.asset_editor.client.compose.lib.action

import fr.hardel.asset_editor.network.workspace.WorkspaceElementSnapshot
import fr.hardel.asset_editor.network.workspace.WorkspaceSyncPayload
import net.minecraft.resources.Identifier

interface WorkspaceSyncGateway {
    fun handleWorkspaceSync(payload: WorkspaceSyncPayload)

    fun handlePackWorkspaceSync(
        packId: String,
        registryId: Identifier,
        snapshots: List<WorkspaceElementSnapshot>
    )
}
